spk/sonarr/src/service-setup.sh

```diff
@@ -43,62 +36,69 @@ SVC_WAIT_TIMEOUT=90
 
 service_postinst ()
 {
-    mkdir -p ${CONFIG_DIR}
-    set_unix_permissions "${CONFIG_DIR}"
+    mkdir -p "${SONARR_CONFIG_DIR}"
+    
+    echo "Set update required"
+    # Make Sonarr do an update check on start to avoid possible Sonarr
+    # downgrade when synocommunity package is updated
+    touch "${SONARR_CONFIG_DIR}/update_required"
```

Comment:
this requires the config file to be set to True for update automatically to function.

```csharp
                // Don't do a prestartup update check unless BuiltIn update is enabled
                if (!_configFileProvider.UpdateAutomatically ||
                    _configFileProvider.UpdateMechanism != UpdateMechanism.BuiltIn ||
                    _deploymentInfoProvider.IsExternalUpdateMechanism)
                {
                    _logger.Debug("Built-in updater disabled, skipping post-install update check");
                    return;
                }
```

https://github.com/Sonarr/Sonarr/blob/main/src/NzbDrone.Core/Update/InstallUpdateService.cs#L304-L311

the config file default is false

```csharp
        public bool UpdateAutomatically => GetValueBoolean("UpdateAutomatically", false, false);
```
https://github.com/Sonarr/Sonarr/blob/e4c0e80e3e12688a2712078b65d9693c9e3d5c8f/src/NzbDrone.Core/Configuration/ConfigFileProvider.cs#L223